### PR TITLE
enable golangci linters, fix linter errors

### LINKED ISSUE #206: fix linter issues
activate linters, and fix issues

### PR DESCRIPTION
**Pull request by lpuskas**
* enable all linters
* temporarily set max line length to 200 cars (lll)
* temporarily disable the go-critic linter (the reported error is not trivial to fix)

fixes #206

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+linters-settings:
+    golint:
+        min-confidence: 0.9
+
+    gocyclo:
+        min-complexity: 15
+
+linters:
+    enable:
```
Comment (sagikazarmark) [inline .golangci.yml]: `enable-all: true` would be better

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        - gofmt
+        - golint
+        - goimports
+        - gochecknoglobals
+    disable:
+        - errcheck
+        - gas
+        - megacheck
+        - varcheck
+    fast: false
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

Fast is false by default

File: .golangci.yml (hunk 1 above)
Comment (lpuskas) [reply .golangci.yml]: thanx, done

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    gocyclo:
+        min-complexity: 20
+
+    lll:
+        line-length: 200
+
+linters:
+    enable-all: true
+    #        - misspell
```
Comment (sagikazarmark) [inline .golangci.yml]: Y tho?

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+linters-settings:
+    golint:
+        min-confidence: 0.9
+
+    gocyclo:
+        min-complexity: 20
+
+    lll:
+        line-length: 200
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
        line-length: 200

    goimports:
        local-prefixes: github.com/sagikazarmark/modern-go-application
```

File: .golangci.yml (hunk 3 above)
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+linters:
+    enable-all: true
+    #        - misspell
+    #        - gofmt
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    #        - golint
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    #        - goimports
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    #        - gochecknoglobals
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion

```

File: .golangci.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    disable:
+        - gocritic
+#        - errcheck
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+#        - gas
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+#        - megacheck
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+#        - varcheck
```
Comment (sagikazarmark) [inline .golangci.yml]:
```suggestion
```

File: .golangci.yml (hunk 4 above)
Comment (lpuskas) [reply .golangci.yml]: thx, fixed

**Pull request merged**